torch/_decomp/decompositions.py

```diff
@@ -3993,7 +3993,7 @@ def scaled_dot_product_flash_attention(
         query, key, value, attn_mask, dropout_p, is_causal, None, scale=scale
     )
     return (
-        output,
+        output.transpose(1, 2),
```

Comment:
oh wow this is so weird

Reply:
it is indeed. flash attnetion, https://fburl.com/38pwyabk, does that while the other one does not. I am gonna check if i trigger non flash variant what happens. But surprising calls to contiguous, https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L5441, does not appear in trace. which is also very weird.

Reply:
Is it a bug? It would be great to add this explanation in the comment. What's the relationship to decomposition statement in the summary?